### PR TITLE
Proposal to only generate OmniBOR when OMNIBOR_DIR specified.

### PR DESCRIPTION
**Pull request by edwarnicke**
Signed-off-by: Ed Warnicke <hagbard@gmail.com>

### CODE REVIEW

**Comment by jsgf** [comment]
Two thoughts:
- Writing this kind of metadata should be opt-in - tools shouldn't be writing extraneous data without being asked to. 
- Even if it is writing the metadata, the tool can't reliably guess where it should write. " A subdirectory ```.adg/``` of the directory to which it is writing out the artifact" assumes that all output artifacts are being written to the same dir, which is not necessarily the case. It should be explicitly configured either via env var, command line option, etc.

File: SPEC-annex-A-filesystem-storage.md [hunk 1]
```diff
@@ -42,8 +42,5 @@ Subdirectory structure, filenaming, and file schema below that point are at the
 OMNIBOR_DIR should be set in order of precedence by:
 1.  A non-empty env variable named OMNIBOR_DIR
 2.  A build tool specific flag
-3.  A subdirectory ```.adg/``` of the directory to which it is writing out the artifact
-
-- The presence of an empty OMNIBOR_DIR env var should be taken as a signal to skip OmniBOR generation
-- The lack of a OMNIBOR_DIR env var should not be taken as a signal to _not_ generate a OmniBOR.
 
+- The absense of specification of an OMNIBOR_DIR should be taken as a signal to skip OmniBOR generation.
```
Comment (bharsesh) [inline SPEC-annex-A-filesystem-storage.md:46]: *absence

**Review by bharsesh — APPROVED** [review]

**Review by yonhan3 — APPROVED** [review]

File: SPEC-annex-A-filesystem-storage.md [hunk 2]
```diff
@@ -42,8 +42,5 @@ Subdirectory structure, filenaming, and file schema below that point are at the
 OMNIBOR_DIR should be set in order of precedence by:
 1.  A non-empty env variable named OMNIBOR_DIR
 2.  A build tool specific flag
-3.  A subdirectory ```.adg/``` of the directory to which it is writing out the artifact
-
-- The presence of an empty OMNIBOR_DIR env var should be taken as a signal to skip OmniBOR generation
-- The lack of a OMNIBOR_DIR env var should not be taken as a signal to _not_ generate a OmniBOR.
 
+- The absence of specification of an OMNIBOR_DIR should be taken as a signal to skip OmniBOR generation.
```
Comment (edwarnicke) [inline SPEC-annex-A-filesystem-storage.md:46]:
```suggestion
The absence of specification of a location to write omnibor data via either the OMNIBOR_DIR or the build tool specific flag  should be taken as a signal to skip OmniBOR generation.
```

**Comment by alilleybrinker** [comment]
- We may want to specify a recommended way to set `OMNIBOR_DIR` to make use of the XDG base directory spec, for those who want to follow the spec. Something like a recommendation to set `export OMNIBOR_DIR="${XDG_DATA_HOME}/omnibor/".
- Do we want to specify that `OMNIBOR_DIR` may be validly set with or without a trailing path separator?

File: SPEC-annex-A-filesystem-storage.md [hunk 3]
```diff
@@ -42,8 +42,5 @@ Subdirectory structure, filenaming, and file schema below that point are at the
 OMNIBOR_DIR should be set in order of precedence by:
```
Comment (edwarnicke) [inline SPEC-annex-A-filesystem-storage.md:42]:
```suggestion
OMNIBOR_DIR should be set by one of:
```

File: SPEC-annex-A-filesystem-storage.md [hunk 4]
```diff
@@ -42,8 +42,5 @@ Subdirectory structure, filenaming, and file schema below that point are at the
...
 1.  A non-empty env variable named OMNIBOR_DIR
 2.  A build tool specific flag
```
Comment (edwarnicke) [inline SPEC-annex-A-filesystem-storage.md:44]: 1. A build tool specific flag
2. A non-empty env variable named OMNIBOR_DIR

File: SPEC-annex-A-filesystem-storage.md (hunk 3 above)
Comment (AevaOnline) [inline SPEC-annex-A-filesystem-storage.md:42]:
```suggestion
OMNIBOR_DIR may be set by the following methods, listed in order of precedence:
```

**Review by fkautz — APPROVED** [review]

**Pull request merged**